### PR TITLE
[terra-table] Allow consumers to hide table column headers

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The Terra Table component was updated to allow consumers to control the visibility of column headers in the table.

**Why it was changed:**
The change was made to maintain passivity with existing implementations that requires the column headers to be hidden such as the Stella Timeline.

### Testing

This change was tested using:

- [X] WDIO
- [X] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [X] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details
N/A

**This PR resolves:**

UXPLATFORM-9638 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/doc/table/Examples/TableWithoutHeaders.jsx
Comment (adoroshk) [inline packages/terra-framework-docs/src/terra-dev-site/doc/table/Examples/TableWithoutHeaders.jsx:16]: Is the `isSelectable` here by accident, or does it have some effect?

File: packages/terra-table/src/Table.jsx
Comment (adoroshk) [inline packages/terra-table/src/Table.jsx:311]: I guess we need to add a `key` prop:
Screenshot 2023-10-19 at 10 57 51 AM

File: packages/terra-framework-docs/src/terra-dev-site/doc/table/Examples/TableWithoutHeaders.jsx
Comment (cm9361) [reply packages/terra-framework-docs/src/terra-dev-site/doc/table/Examples/TableWithoutHeaders.jsx:16]: We are setting the value to false.  This will make it not selectable.

**Review by adoroshk — APPROVED** [review]

**Review by kenk2 — APPROVED** [review]

**Comment by chrismichalewicz** [comment]
@cm9361  Functionally it looks good, confirmed no headers are read in JAWS as well.

**Review by eawww — COMMENTED** [review]
This wasn't clear in the jira but for the Stella Timeline case where the headers need to be visually hidden, they still need to exist to give that context to screen readers. A screen reader won't know to associate the external timeline as the column headers.

**Review by sdadn — APPROVED** [review]

**Comment by eawww** [comment]
It gets the header info now! Thanks Charles. One last thing is the header cells shouldn't have buttons since they can't be accessed by anyone without a screen reader.

**Comment by cm9361** [comment]
> It gets the header info now! Thanks Charles. One last thing is the header cells shouldn't have buttons since they can't be accessed by anyone without a screen reader.

Thank you for the feedback, @eawww.  I need the logic from upcoming PRs to remove the button.  I would not want to add that to this PR to prevent merge conflicts.  What I can do is make the cell be 'not selectable'.  This should give the desired behavior once the JIRAs in-flight are merged.  Would that be acceptable?

**Comment by eawww** [comment]
@cm9361 could you point me to the jira or PR removal of the button is slated on?

**Comment by cm9361** [comment]
> @cm9361 could you point me to the jira or PR removal of the button is slated on?

https://github.com/cerner/terra-framework/pull/1836/files#diff-6e67623f4efab392deb44bd2b292882500459c61651ae61d911cd8dbd01bc775R271

**Review by eawww — APPROVED** [review]
🥶 Cool beans! 🫘
Approving with the understanding that the removal of the button is being accomplished elsewhere.

**Pull request merged**
